### PR TITLE
Update docker image automatically

### PR DESCRIPTION
**Pull request by gal-berger**
## Status
- [] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/28475

## Description
Running update RN command when Docker image has changed add a record for it in RN.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: demisto_sdk/commands/update_release_notes/update_rn.py [hunk 1]
```diff
@@ -102,6 +104,8 @@ def execute_update(self):
                 if self.is_bump_required():
                     self.commit_to_bump(new_metadata)
                 self.create_markdown(rn_path, rn_string, changed_files)
+                if is_docker_image_changed:
+                    self.update_markdown(rn_path, f'- Upgraded the Docker image to {docker_image_name}')
```
Comment (yaakovi) [inline demisto_sdk/commands/update_release_notes/update_rn.py:108]:
```suggestion
                    self.update_markdown(rn_path, f'- Upgraded the Docker image to {docker_image_name}.')
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -26,6 +26,7 @@
 * Added support for *prev-ver* flag in **lint** and **secrets** commands.
 * Added support for *text* flag to **update-release-notes** command to add the same text to all release notes.
 * Fixed an issue where **validate** did not recognize added files if they were modified locally.
+* Add enhancement to **update-release-notes**: update release notes automatically when upgrading docker image.
```
Comment (roysagi) [inline CHANGELOG.md:29]:
```suggestion
* The **update-release-notes** command will now automatically add docker image update to the release notes.
```

File: demisto_sdk/commands/update_release_notes/update_rn.py [hunk 3]
```diff
@@ -88,9 +88,11 @@ def execute_update(self):
             self.check_rn_dir(rn_path)
             changed_files = {}
             self.find_added_pack_files()
+            is_docker_image_changed = False
             for packfile in self.modified_files_in_pack:
                 file_name, file_type = self.identify_changed_file_type(packfile)
-
+                if 'yml' in packfile:
```
Comment (roysagi) [inline demisto_sdk/commands/update_release_notes/update_rn.py:94]: This should be done by `file_type` - that way you won't check playbooks

File: demisto_sdk/commands/update_release_notes/update_rn.py [hunk 4]
```diff
@@ -526,3 +539,19 @@ def update_api_modules_dependents_rn(_pack, pre_release, update_type, added, mod
...
+
+
+def check_docker_image_changed(added_or_modified_yml):
+    try:
+        diff = run_command(f'git diff origin/master -- {added_or_modified_yml}', exit_on_error=False)
+    except RuntimeError as e:
+        if any(['is outside repository' in exp for exp in e.args]):
+            return False, ''
+        else:
+            raise
```
Comment (roysagi) [inline demisto_sdk/commands/update_release_notes/update_rn.py:551]: dont raise the error - print something to the user about being unable to check the docker image and skipping

File: demisto_sdk/commands/update_release_notes/update_rn.py [hunk 5]
```diff
@@ -526,3 +539,19 @@ def update_api_modules_dependents_rn(_pack, pre_release, update_type, added, mod
...
+    else:
+        diff_lines = diff.splitlines()
+        for diff_line in diff_lines:
+            if '+  dockerimage:' in diff_line:
```
Comment (roysagi) [inline demisto_sdk/commands/update_release_notes/update_rn.py:555]: cool idea -please add a note explaining the `+ dockerimage`

**Review by roysagi — APPROVED** [review]
Please address my notes.

Looks good.

File: demisto_sdk/commands/update_release_notes/update_rn.py (hunk 3 above)
Comment (gal-berger) [reply demisto_sdk/commands/update_release_notes/update_rn.py:94]: good idea.
I'll add it.

**Review by yuvalbenshalom — APPROVED** [review]

**Pull request merged**